### PR TITLE
Upgrade user group documentation

### PR DESCRIPTION
**Pull request by rathodmeetsatish**

### CODE REVIEW

**Review by RajBaviskar — APPROVED** [review]

File: docs/resources/platform_usergroup.md [hunk 1]
```diff
@@ -3,17 +3,73 @@
 page_title: "harness_platform_usergroup Resource - terraform-provider-harness"
 subcategory: "Next Gen"
 description: |-
-  Resource for creating a Harness User Group.
+  Resource for creating a Harness User Group.Linking SSO providers with User Groups:
+  
+      When the SSO Provider is of LDAP type, the following values need to be populated:
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:8]: The following fields need to be populated for LDAP SSO Providers:

File: docs/resources/platform_usergroup.md [hunk 2]
```diff
@@ -3,17 +3,73 @@
...
+  
+      1) linked_sso_id
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:10]: Is there a sequence for this? If not, we can remove the numbering and make it a bulleted list.

File: docs/resources/platform_usergroup.md [hunk 3]
```diff
@@ -3,17 +3,73 @@
...
+  
+      3) sso_group_id
+  
+      4) sso_group_name
+  
+      5) linked_sso_type
+  
+      6) sso_linked
+  
+      When the SSO Provider is of SAML type, the following values need to be populated:
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:22]: The following fields need to be populated for SAML SSO Providers:

File: docs/resources/platform_usergroup.md [hunk 4]
```diff
@@ -3,17 +3,73 @@
...
+  
+      5) linked_sso_type
+  
+      6) sso_linked
 ---
 
 # harness_platform_usergroup (Resource)
 
-Resource for creating a Harness User Group.
+Resource for creating a Harness User Group.Linking SSO providers with User Groups:
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:39]: Space is missing after "User Group."

File: docs/resources/platform_usergroup.md [hunk 5]
```diff
@@ -3,17 +3,73 @@
...
+
+		When the SSO Provider is of LDAP type, the following values need to be populated:
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:41]: Refer to the previous comments.

File: docs/resources/platform_usergroup.md [hunk 6]
```diff
@@ -3,17 +3,73 @@
...
+		
+		1) linked_sso_id
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:43]: Refer to the previous comments.

File: docs/resources/platform_usergroup.md [hunk 7]
```diff
@@ -3,17 +3,73 @@
...
+		
+		3) sso_group_id
+		
+		4) sso_group_name
+		
+		5) linked_sso_type
+		
+		6) sso_linked
+		
+		When the SSO Provider is of SAML type, the following values need to be populated:
```
Comment (rashminandasahoo) [inline docs/resources/platform_usergroup.md:55]: Refer to the previous comments.

File: internal/service/platform/usergroup/usergroup.go [hunk 8]
```diff
@@ -13,7 +14,36 @@ import (
...
+		
+		3) sso_group_id
+		
+		4) sso_group_name
+		
+		5) linked_sso_type
+		
+		6) sso_linked
+		
+		When the SSO Provider is of SAML type, the following values need to be populated:
```
Comment (rashminandasahoo) [inline internal/service/platform/usergroup/usergroup.go:34]: Refer to the previous comments.

**Review by rashminandasahoo — CHANGES_REQUESTED** [review]

File: docs/resources/platform_usergroup.md [hunk 9]
```diff
@@ -3,17 +3,73 @@
 page_title: "harness_platform_usergroup Resource - terraform-provider-harness"
 subcategory: "Next Gen"
 description: |-
-  Resource for creating a Harness User Group.
+  Resource for creating a Harness User Group.Linking SSO providers with User Groups:
```
[the remark this answers is not in the document]
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 1 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 2 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 3 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 4 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 5 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 6 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: docs/resources/platform_usergroup.md (hunk 7 above)
Comment (rathodmeetsatish) [reply docs/resources/platform_usergroup.md]: done

File: internal/service/platform/usergroup/usergroup.go (hunk 8 above)
Comment (rathodmeetsatish) [reply internal/service/platform/usergroup/usergroup.go]: done

**Review by rashminandasahoo — APPROVED** [review]

**Pull request merged**